contracts/hardhat.config.ts

```diff
@@ -56,27 +92,7 @@ const config: HardhatUserConfig = {
       url: "http://localhost:8545",
       loggingEnabled: false,
     },
-    hardhat: {
-      blockGasLimit: DEFAULT_BLOCK_GAS_LIMIT,
-      gasMultiplier: DEFAULT_GAS_MUL,
-      gasPrice: NETWORKS_DEFAULT_GAS[ESupportedChains.Hardhat],
-      chainId: EChainId.Hardhat,
-      accounts: {
-        mnemonic: TEST_MNEMONIC,
-        path: "m/44'/60'/0'/0",
-        initialIndex: 0,
-        count: 20,
-      },
-      gas: DEFAULT_BLOCK_GAS_LIMIT,
-      loggingEnabled: false,
-      allowUnlimitedContractSize: true,
-      throwOnTransactionFailures: true,
-      throwOnCallFailures: true,
-      mining: {
-        auto: true,
-        interval: 100,
-      },
-    },
+    hardhat: getHardhatConfig(),
```

Comment:
Why do you need this function? Forking can be added without changing this config logic.

```
    hardhat: {
      blockGasLimit: DEFAULT_BLOCK_GAS_LIMIT,
      gasMultiplier: DEFAULT_GAS_MUL,
      gasPrice: NETWORKS_DEFAULT_GAS[ESupportedChains.Hardhat],
      chainId: EChainId.Hardhat,
      accounts: {
        mnemonic: TEST_MNEMONIC,
        path: "m/44'/60'/0'/0",
        initialIndex: 0,
        count: 20,
      },
      gas: DEFAULT_BLOCK_GAS_LIMIT,
      loggingEnabled: false,
      allowUnlimitedContractSize: true,
      throwOnTransactionFailures: true,
      throwOnCallFailures: true,
      mining: {
        auto: true,
        interval: 100,
      },
      forkings: process.env.FORKING_URL ? { ... } : undefined,
    },
```

Reply:
just think it would be clear? but this way is also good 👍